lib/user/user-client.js

```diff
@@ -29,13 +31,65 @@ var xhr = function(_options) {
   return deferred.promise;
 };
 
-var storeProfile = function(profileData) {
-  localStorage.setItem('fh.wfm.profileData', JSON.stringify(profileData));
+/**
+* Encrypt the users profile data using AES with a hash of the session key and store the result in localStorage.
+* @param profileData {object} - the plaintext to encrypt.
+* @param sessionToken {string} - the secret key to encrypt with.
+*/
+var storeProfile = function(profileData, sessionToken) {
+  if (sessionToken && profileData) {
+
+    var hashedSessionKey = sha256.hash(sessionToken);
+    // handle failed hashing of session token
+    if (hashedSessionKey instanceof Error) {
+      storeLocalStorage('fh.wfm.profileData', null, hashedSessionKey.toString());
+      return;
+    }
+
+    var encryptedData = aes.encrypt(profileData, hashedSessionKey);
```

Comment:
Please please please don't return errors. It breaks the traditional flow of code, requires you to manually do `if (result instanceof Error)` every time you call it, and if you forget to do that check, you'll likely get an indecipherable runtime error when you try to use the result as a string(?) and it's actually an Error. (Think of trying to pass an Error object to a Buffer constructor. `TypeError: First argument must be a string, Buffer, ArrayBuffer, Array, or array-like object.`)

Throwing is the way to go. Always. (Unless you're using async callbacks, of course.)